src/oidcc_backoff.erl

```diff
@@ -0,0 +1,47 @@
+%%%-------------------------------------------------------------------
+%% @doc Backoff Handling
+%%
+%% Based on `db_connection':
+%% [https://github.com/elixir-ecto/db_connection/blob/8ef1f2ea54922873590b8939f2dad6b031c5b49c/lib/db_connection/backoff.ex#L24]
+%% @end
+%% @since 3.2.0
+%%%-------------------------------------------------------------------
+-module(oidcc_backoff).
+
+-export_type([type/0]).
+-export_type([min/0]).
+-export_type([max/0]).
+-export_type([state/0]).
+
+-export([handle_retry/4]).
+
+-type type() :: stop | exponential | random | random_exponential.
+
+-type min() :: pos_integer().
```

Comment:
question(non-blocking): this can't be 0? if it can be 0:
```suggestion
-type min() :: non_neg_integer().
```

Reply:
I don't think a backoff 0 makes sense. When would you set it to 0?

Some strategies would also not make sense with 0. You can multiply 0 infinite times and still have zero. So the exponential strategy would create an infinite loop.

Reply:
@paulswartz Some guards definitely make sense. I‘ll add some.

For immediate retry, stop should work fine.